### PR TITLE
Bump Golang 1.9.2

### PR DESCRIPTION
**Pull request by itscaro**
- Golang 1.9.2
- multi-line in Dockerfile for readability
- add `build_samples` in Makefile

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -16,7 +16,7 @@ before_script:
 script:
   - go get -d -v
```
Comment (stefanprodan) [inline .travis.yml]: I think we don't need to run `go get` since we have the vendor dir on git.

Comment (alexellis) [reply .travis.yml]: Agreed we can remove that line now.

File: Dockerfile [hunk 2]
```diff
@@ -17,12 +19,18 @@ RUN license-check -path ./ --verbose=false \
...
  && VERSION=$(git describe --all --exact-match `git rev-parse HEAD` | grep tags | sed 's/tags\///') \
  && GIT_COMMIT=$(git rev-list -1 HEAD) \
- && CGO_ENABLED=0 GOOS=linux go build --ldflags "-s -w -X github.com/openfaas/faas-cli/version.GitCommit=${GIT_COMMIT} -X github.com/openfaas/faas-cli/version.Version=${VERSION}" -a -installsuffix cgo -o faas-cli
+ && CGO_ENABLED=0 GOOS=linux go build --ldflags "-s -w \
+    -X github.com/openfaas/faas-cli/version.GitCommit=${GIT_COMMIT} \
+    -X github.com/openfaas/faas-cli/version.Version=${VERSION}" \
+    -a -installsuffix cgo -o faas-cli
 
+# Release stage
 FROM alpine:latest
```
Comment (alexellis) [inline Dockerfile]: Please can you change `latest` to `3.6`?

File: Dockerfile [hunk 3]
```diff
@@ -1,4 +1,6 @@
-FROM golang:1.8.3
+# Build stage
+FROM golang:1.9.2
```
Comment (alexellis) [inline Dockerfile]: Please add `as build` to this line

File: Dockerfile [hunk 4]
```diff
@@ -17,12 +19,18 @@ RUN license-check -path ./ --verbose=false \
...
 
+# Release stage
 FROM alpine:latest
+
 RUN apk --no-cache add ca-certificates
 
 WORKDIR /root/
 
-COPY --from=0 /go/src/github.com/openfaas/faas-cli/faas-cli               . 
+COPY --from=0 /go/src/github.com/openfaas/faas-cli/faas-cli               .
```
Comment (alexellis) [inline Dockerfile]: Let's change `from=0` to `from=builder`

File: Dockerfile.redist [hunk 5]
```diff
@@ -17,13 +19,30 @@ RUN license-check -path ./ --verbose=false \
...
+        -X github.com/openfaas/faas-cli/version.GitCommit=${GIT_COMMIT} \
+        -X github.com/openfaas/faas-cli/version.Version=${VERSION}" \
+        -a -installsuffix cgo -o faas-cli-armhf \
+ && CGO_ENABLED=0 GOOS=linux GOARCH=arm64 go build --ldflags "-s -w \
+        -X github.com/openfaas/faas-cli/version.GitCommit=${GIT_COMMIT} \
+        -X github.com/openfaas/faas-cli/version.Version=${VERSION}" \
+        -a -installsuffix cgo -o faas-cli-arm64
+
+# Release stage
 FROM alpine:latest
```
Comment (alexellis) [inline Dockerfile.redist]: Please use `3.6`

File: Dockerfile (hunk 3 above)
Comment (itscaro) [reply Dockerfile]: as build or builder?

**Comment by alexellis** [comment]
Thanks for the review @stefanprodan - @itscaro thanks for working on this.

**Pull request merged**